_ruby_libs/rosindex.rb

```diff
@@ -79,7 +79,7 @@ def initialize(name, type, uri, purpose, checkout_path, id = nil)
 
     # hash distro -> RepoSnapshot
     # each entry in this hash represents the preferred version for a given distro in this repo
-    @snapshots = Hash[$all_distros.collect { |d| [d, RepoSnapshot.new(nil, d, false, false)] }]
+    @snapshots = {}
```

Comment:
Is this the beginning of the orphaned code cleanup? Should we maybe defer that to later? I see that we're still using the snapshots in the main loop: https://github.com/ros-infrastructure/rosindex/pull/352/files#diff-70c045b5c7c6b1ab5c817acfead858a20af39af4e4724acc3c43caa0c0b20497R698

Reply:
This code adds a blank repo snapshot that matches all existing distro names, when a Repo instance is created. Later, when the release data is scanned, the snapshots for official distros are replaced with correct data. Then eventually all of these blank instances are aggressively searched for, trying to match a branch with the distro names. It is these searched-for instances that this PR is eliminating. AFAICT, there is no use for these blank instances other than to feed the aggressive search for the instances, that this PR is eliminating.

So yes this is orphaned code in the sense that later I could tell the difference between an officially released and a searched-for snapshot, and not use the searched-for instances. But I don't see the point in leaving blank instances here that are not used for anything (unless I am mistaken in their use).